www/src/data/sidebars/doc-links.yaml

```diff
@@ -538,6 +538,8 @@
           link: /docs/building-with-components/
         - title: Lifecycle APIs
           link: /docs/gatsby-lifecycle-apis/
+        - title: React Rehydration
```

Comment:
Wasn't sure of the best place to put this, I thought about the Internals section under Webpack and SSR but it didn't feel like it was going to go as "in depth" so I opted for the Conceptual Guide section.